lib/spack/spack/installer.py

```diff
@@ -901,7 +924,7 @@ def __init__(
         installed: Set[str],
     ):
         """
-        Instantiate a build task for a package.
+        Instantiate a task for a package.
```

Comment:
This is not necessarily to be done in this PR, but the `attempts` argument is unused. Another change for a following PR that might improve readability is to force using keyword arguments for arguments after `pkg`. This is because, when reading a call:
```python
BuildTask(pkg, request, False, 0, 0, inst.STATUS_ADDED, set())
```
it's kind of hard to understand what's being passed without constantly jumping to the definition.

Reply:
The `attempts` argument is technically used, although it's only to determine whether to print `Queueing` or `Requeueing` in a status update message.

Reply:
Hmm, it seems to me the `Task.attempts` _attribute_ is used but the `__init__` argument is not:

https://github.com/spack/spack/blob/6f5f3d862ccc1e2472c3c77552f626dc13bcd8c6/lib/spack/spack/installer.py#L1000

Reply:
@alalazo It is *supposed* to be used.  I have a PR to address this (#46293).  

After discussing with @becker33, I'll make the kwargs change in that same PR.

Reply:
Resolving this as handled in #46293